### PR TITLE
Transform the same image twice for contrastive learning

### PR DESCRIPTION
**Pull request by asbjrnmunk**
I think this is pretty cool. This allows us to seamlessly use contrastive learning with any of the inbuilt augmentations from Yucca. 

Usage (pseudo code):

```
dataset = YuccaDataset(task_type="contrastive")
loader = DataLoader(
    dataset,
    batch_size=2,
)


for batch in loader:
    x1, x2 = batch['data']
    print(x1.shape)
    print(x2.shape)
```

Should print

```
torch.Size([2, 1, 24, 24])
torch.Size([2, 1, 24, 24])
...
```
for at patch size of 24.

### CODE REVIEW

**Comment by asbjrnmunk** [comment]
I was a bit fast adding you to the review. I have to test the code first :-) Sorry!

**Comment by jakobamb** [comment]
This is really cool! Let me know when you are done, then I'll take a closer look at the code :)

File: yucca/training/data_loading/YuccaDataset.py [hunk 1]
```diff
@@ -11,21 +11,21 @@
...
+        self.all_cases = samples
         self.composed_transforms = composed_transforms
         self.patch_size = patch_size
         self.task_type = task_type
         if label_dtype is None:
             if self.task_type in ["segmentation", "classification"]:
                 self.label_dtype = torch.int32
-            if self.task_type == "unsupervised":
+            if self.task_type in ["unsupervised", "contrastive"]:
                 self.label_dtype = torch.float32
```
Comment (asbjrnmunk) [inline yucca/training/data_loading/YuccaDataset.py:29]: I think this is super confusing and cannot understand why we need it, but code is OOP, so cannot understand how not either.

**Comment by asbjrnmunk** [comment]
@jakobamb its done

File: yucca/training/data_loading/YuccaDataset.py [hunk 2]
```diff
@@ -11,22 +12,21 @@
...
         self.task_type = task_type
-        if label_dtype is None:
-            if self.task_type in ["segmentation", "classification"]:
-                self.label_dtype = torch.int32
-            if self.task_type == "unsupervised":
-                self.label_dtype = torch.float32
+        if label_dtype is None and self.task_type in ["segmentation", "classification"]:
+            self.label_dtype = torch.int32
+        else:
+            self.label_dtype = label_dtype
```
Comment (asbjrnmunk) [inline yucca/training/data_loading/YuccaDataset.py:29]: @Sllambias these changes seem to work and makes a lot more sense to me, but might be wrong, so please let me know if i have misunderstood something! (the old change used float for tasks that did not use labels and did not pass on the `label_dtype` param)

Comment (jakobamb) [reply yucca/training/data_loading/YuccaDataset.py:29]: I would like to have a one line comment here, why could it happen that label_dtype is None in either the segmentation or classification setting?

**Review by jakobamb — APPROVED** [review]
Really cool that we can make it work this elegantly! It is actually not directly related to your changes in this PR, but I think for readability it would make a lot of sense to add a one-line comment for the strange `if label_dtype is None and self.task_type in ["segmentation", "classification"]:` case

**Review by Sllambias — COMMENTED** [review]
Minor comments
